src/components/spin-button/spin-button.component.ts

```diff
@@ -93,4 +95,77 @@ export class SpinButtonComponent implements ControlValueAccessor {
         this.disabled = isDisabled;
     }
 
+    onKeypress(event: KeyboardEvent): boolean {
+
+        // we only need to perform checks if the type is number
+        if (this.type !== 'number') {
+            return;
+        }
+
+        if (!this._regex.test(event.key)) {
```

Comment:
I don't think this should be the same regex as the one below. `event.key` will only ever be a single character so I believe it would cause problems when trying to type the "-" of "-1".

Here's a plunker example: https://plnkr.co/edit/BzKEf820MuSF6hKmfoRZ?p=preview